### PR TITLE
Adds Webpack Renderer

### PR DESCRIPTION
**Pull request by danielcherubini**
This is a big one.. it's probably best to pull it and run it locally

### CODE REVIEW

File: lib/renderer/renderer-vueify.js [hunk 1]
```diff
@@ -77,7 +79,10 @@ class Renderer {
             }
         }
         this.rootPath = Utils.FindRootPath(options.rootPath);
-
+        this.babel = options.babel;
+        if (options.babel) {
+            vueify.compiler.applyConfig({ babel: options.babel});
+        }
         if (!process.env.VUE_DEV) {
```
Comment (nikos-fytros) [inline lib/renderer/renderer-vueify.js:86]: This is a string so if I set it to false it won't trigger. Also why don't we run that everytime on app startup? That will build everything and the BuildConfig should not be called again
```suggestion
        if (!process.env.VUE_DEV || (process.env.VUE_DEV && process.env.VUE_DEV === "false")) {
```

File: lib/renderer/renderer-webpack.js [hunk 2]
```diff
@@ -0,0 +1,524 @@
...
+                                                console.error(stat.compilation.errors);
+                                            }
+                                        });
+                                    } else if (stats.hasErrors()) {
+                                        console.error(stats.compilation.errors);
+                                    }
+                                }
+                                console.debug(`Cached ${filepath}`);
+                                // we use a counter because it's an async function and can go out of sync
+                                count++;
```
Comment (nikos-fytros) [inline lib/renderer/renderer-webpack.js]: Can't we use the index var that we already have? Why need a new one? Is that counter only for logging?

File: lib/renderer/renderer-webpack.js [hunk 3]
```diff
@@ -0,0 +1,524 @@
...
+                console.error(`Error Precacheing ${filepath} \n\n ${error}`);
+            }
+        }
+    }
+    /**
+     * @param {string} filePath
+     * @param {string} vueFile
+     * @returns {Promise<WebpackConfigType>}
+     */
+    async BuildConfig(filePath, vueFile) {
```
Comment (nikos-fytros) [inline lib/renderer/renderer-webpack.js]: Does this run for all .vue files? It should not run for components. They have another component as a root and it doesn't make sense to instantiate a new vue class for those

File: lib/renderer/renderer-webpack.js [hunk 4]
```diff
@@ -0,0 +1,524 @@
...
+            app.$mount("#app")`;
+
+            const parsedFilePath = filePath.replace(this.trueRootPath, "").replace(".vue", "");
+            const configFolder = path.join(this.trueRootPath, ".expressvue", parsedFilePath);
+            try {
+                fs.statSync(configFolder).isDirectory();
+            } catch (error) {
+                mkdirp.sync(configFolder);
+            }
+
```
Comment (nikos-fytros) [inline lib/renderer/renderer-webpack.js:140]: Shouldn't all the config options be set once on project startup?

File: lib/renderer/renderer-webpack.js [hunk 5]
```diff
@@ -0,0 +1,524 @@
...
+            const appPath = path.join(configFolder, "app.js");
+            const serverPath = path.join(configFolder, "entry-server.js");
+            const clientPath = path.join(configFolder, "entry-client.js");
+
+            fs.writeFileSync(appPath, app, "utf-8");
```
Comment (nikos-fytros) [inline lib/renderer/renderer-webpack.js]: Why write file sync and not async?
```suggestion
            await Promise.all([fs.writeFile(appPath, app, "utf-8"),fs.writeFile(serverPath, server, "utf-8"), fs.writeFile(clientPath, client, "utf-8")])
```

**Review by nikos-fytros — COMMENTED** [review]
Have some comment

File: lib/renderer/renderer-webpack.js (hunk 5 above)
Comment (danielcherubini) [reply lib/renderer/renderer-webpack.js]: `fs.writeFile` takes a callback, it's not a promise.

the `fs.promises` is still new and experimental, and only in node 10

**Review by nikos-fytros — APPROVED** [review]
Let's publish it in a new major and try it in one of our projects

**Comment by danielcherubini** [comment]
@nick-fytros you mean minor.. since this isn't a deprecation.. 

what i can do is just do it locally, or use a pre-release

**Pull request merged**